lib/jerboa/format/body/attribute/error_code.ex

```diff
@@ -0,0 +1,143 @@
+defmodule Jerboa.Format.Body.Attribute.ErrorCode do
+  @moduledoc """
+  ERROR-CODE attribute as defined in [STUN RFC](https://tools.ietf.org/html/rfc5389#section-15.6)
+  """
+
+  alias Jerboa.Format.Body.Attribute.{Decoder,Encoder}
+  alias Jerboa.Format.ErrorCode.{FormatError, LengthError}
+  alias Jerboa.Params
+
+  defstruct [:code, :name, reason: ""]
+
+  @typedoc """
+  Represents error code of error response
+
+  Struct fields
+  * `:code` - integer representation of an error
+  * `:name` - atom representation of an error
+  * `:reason`
+  """
+  @type t :: %__MODULE__{
+    code: code,
+    name: name,
+    reason: String.t
+  }
+
+  @type code :: 300 | 400 | 401 | 420 | 438 | 500
+              | 403 | 437 | 441 | 442 | 486 | 508
+
+  @type name :: :try_alternate
+              | :bad_request
+              | :unauthorized
+              | :unknown_attribute
+              | :stale_nonce
+              | :server_error
+              | :forbidden
+              | :allocation_mismatch
+              | :wrong_credentials
+              | :unsupported_protocol
+              | :allocation_quota_reached
+              | :insufficient_capacity
+
+  @valid_codes [300, 400, 401, 420, 438, 500,
```

Comment:
I'm wondering if there's a way not to repeat all the types of the `@code` in the `@valid_codes`.